13:51:37,497 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000022A9E146CD0>}).
13:51:37,497 src.helpers.handle_error_decorator - DEBUG - Error Occurred: get_courses Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 19, in extract_token_data
    payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORTIHM])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 157, in decode
    _validate_claims(
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 481, in _validate_claims
    _validate_exp(claims, leeway=leeway)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 314, in _validate_exp
    raise ExpiredSignatureError("Signature has expired.")
jose.exceptions.ExpiredSignatureError: Signature has expired.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\setup_logger.py", line 22, in wrapper
    val = fun(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 44, in get_courses
    jwt_token_data = extract_token_data(request)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 29, in extract_token_data
    raise HTTPException(status_code=404, detail=str(error))
fastapi.exceptions.HTTPException

13:51:37,497 src.helpers.handle_error_decorator - ERROR - Error Occurred: get_courses Method Error: 
13:51:46,474 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000022A9E0E62D0>}).
13:51:46,482 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price': 2000, 'rating (in Rs.)': 4.77}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price': 1000, 'rating (in Rs.)': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price': 1500, 'rating (in Rs.)': 0.0}].
13:52:02,489 src.helpers.setup_logger - DEBUG - mentor_earning called with ((), {'request': <starlette.requests.Request object at 0x0000022A9E1AFDD0>}).
13:52:02,535 src.helpers.setup_logger - DEBUG - mentor_earning returned [{'name': 'abhishek', 'course_name': 'Master dsa', 'earning': 3000}, {'name': 'abhishek', 'course_name': 'Java script', 'earning': 32000}, {'name': 'abhishek', 'course_name': 'clean code', 'earning': 3000}, {'name': 'abhishek', 'course_name': 'dsa master', 'earning': 0}, {'name': 'abhishek', 'course_name': 'dsa ', 'earning': 0}].
14:07:53,572 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 229, in add_course_feedback
    content = course.list_course(4, user_id)
              ^^^^^^
UnboundLocalError: cannot access local variable 'course' where it is not associated with a value

14:07:53,572 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: cannot access local variable 'course' where it is not associated with a value
14:08:31,310 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 229, in add_course_feedback
    content = course.list_course(4, user_id)
              ^^^^^^
UnboundLocalError: cannot access local variable 'course' where it is not associated with a value

14:08:31,310 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: cannot access local variable 'course' where it is not associated with a value
14:09:09,799 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 229, in add_course_feedback
    content = course.list_course(4, user_id)
              ^^^^^^
UnboundLocalError: cannot access local variable 'course' where it is not associated with a value

14:09:09,799 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: cannot access local variable 'course' where it is not associated with a value
14:10:52,931 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 231, in add_course_feedback
    validation_response = validate_request_data(user_feedback, feedback_schema)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\validations.py", line 9, in validate_request_data
    jsonschema.validate(instance=request_data, schema=schema)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1301, in validate
    cls = validator_for(schema)
          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1370, in validator_for
    if schema is True or schema is False or "$schema" not in schema:
                                            ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'module' is not iterable

14:10:52,932 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: argument of type 'module' is not iterable
14:11:29,880 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 232, in add_course_feedback
    validation_response = validate_request_data(user_feedback, feedback_schema)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\validations.py", line 9, in validate_request_data
    jsonschema.validate(instance=request_data, schema=schema)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1301, in validate
    cls = validator_for(schema)
          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1370, in validator_for
    if schema is True or schema is False or "$schema" not in schema:
                                            ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'module' is not iterable

14:11:29,881 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: argument of type 'module' is not iterable
14:11:48,119 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 233, in add_course_feedback
    validation_response = validate_request_data(user_feedback, feedback_schema)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\validations.py", line 9, in validate_request_data
    jsonschema.validate(instance=request_data, schema=schema)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1301, in validate
    cls = validator_for(schema)
          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1370, in validator_for
    if schema is True or schema is False or "$schema" not in schema:
                                            ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'module' is not iterable

14:11:48,120 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: argument of type 'module' is not iterable
14:12:19,819 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 232, in add_course_feedback
    validation_response = validate_request_data(user_feedback, feedback_schema)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\validations.py", line 9, in validate_request_data
    jsonschema.validate(instance=request_data, schema=schema)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1301, in validate
    cls = validator_for(schema)
          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1370, in validator_for
    if schema is True or schema is False or "$schema" not in schema:
                                            ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'module' is not iterable

14:12:19,819 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: argument of type 'module' is not iterable
14:19:30,629 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 231, in add_course_feedback
    validation_response = validate_request_data(user_feedback, feedback_schema)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\validations.py", line 9, in validate_request_data
    jsonschema.validate(instance=request_data, schema=schema)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1301, in validate
    cls = validator_for(schema)
          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1370, in validator_for
    if schema is True or schema is False or "$schema" not in schema:
                                            ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'module' is not iterable

14:19:30,629 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: argument of type 'module' is not iterable
14:19:41,484 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 231, in add_course_feedback
    validation_response = validate_request_data(user_feedback, feedback_schema)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\validations.py", line 9, in validate_request_data
    jsonschema.validate(instance=request_data, schema=schema)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1301, in validate
    cls = validator_for(schema)
          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1370, in validator_for
    if schema is True or schema is False or "$schema" not in schema:
                                            ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'module' is not iterable

14:19:41,485 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: argument of type 'module' is not iterable
14:20:32,429 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_course_feedback Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 233, in add_course_feedback
    validation_response = validate_request_data(user_feedback, feedback_schema)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\validations.py", line 9, in validate_request_data
    jsonschema.validate(instance=request_data, schema=schema)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1301, in validate
    cls = validator_for(schema)
          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jsonschema\validators.py", line 1370, in validator_for
    if schema is True or schema is False or "$schema" not in schema:
                                            ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'module' is not iterable

14:20:32,429 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_course_feedback Method Error: argument of type 'module' is not iterable
14:22:44,284 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000025BF0E0E410>}).
14:22:44,315 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Master dsa', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.65, 'status': 'deactive', 'approval status': 'approved'}, {'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79, 'status': 'active', 'approval status': 'approved'}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75, 'status': 'active', 'approval status': 'approved'}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0, 'status': 'active', 'approval status': 'approved'}, {'name': 'dsa master', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'status': 'active', 'approval status': 'pending'}, {'name': 'dsa ', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'status': 'active', 'approval status': 'pending'}].
14:23:11,877 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000025BF0E0C650>}).
14:23:11,888 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price': 2000, 'rating (in Rs.)': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price': 1000, 'rating (in Rs.)': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price': 1500, 'rating (in Rs.)': 0.0}].
15:01:55,806 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001D0D0B7A090>}).
15:01:55,816 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price': 2000, 'rating (in Rs.)': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price': 1000, 'rating (in Rs.)': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price': 1500, 'rating (in Rs.)': 0.0}].
17:49:53,719 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000024C0CD0D990>}).
17:49:53,751 src.helpers.setup_logger - DEBUG - get_courses returned 1.
17:51:51,197 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002D6F7BEDB10>}).
17:51:51,215 src.helpers.setup_logger - DEBUG - get_courses returned 1.
17:53:54,876 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001A0F2463990>}).
17:53:54,895 src.helpers.setup_logger - DEBUG - get_courses returned 1.
17:54:25,545 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000019A23D1DAD0>}).
17:54:25,563 src.helpers.setup_logger - DEBUG - get_courses returned 1.
17:55:02,695 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000019444825B90>}).
17:55:02,714 src.helpers.setup_logger - DEBUG - get_courses returned 1.
17:55:16,389 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001D6BE3FDE90>}).
17:55:16,407 src.helpers.setup_logger - DEBUG - get_courses returned 1.
17:55:31,608 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001A3F1A61250>}).
17:55:31,635 src.helpers.setup_logger - DEBUG - get_courses returned 'NoneType' object is not iterable.
17:56:21,836 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000017E51FCDB10>}).
17:56:21,855 src.helpers.setup_logger - DEBUG - get_courses returned 1.
17:57:55,215 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001EC7FCCDED0>}).
17:57:55,241 src.helpers.setup_logger - DEBUG - get_courses returned 1.
17:58:10,721 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000186C30CD890>}).
17:58:10,740 src.helpers.setup_logger - DEBUG - get_courses returned 1.
21:53:10,499 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001B84ED4D850>}).
21:53:10,531 src.helpers.handle_error_decorator - DEBUG - Error Occurred: get_courses Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 19, in extract_token_data
    payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORTIHM])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 157, in decode
    _validate_claims(
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 481, in _validate_claims
    _validate_exp(claims, leeway=leeway)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 314, in _validate_exp
    raise ExpiredSignatureError("Signature has expired.")
jose.exceptions.ExpiredSignatureError: Signature has expired.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\setup_logger.py", line 22, in wrapper
    val = fun(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 45, in get_courses
    jwt_token_data = extract_token_data(request)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 29, in extract_token_data
    raise HTTPException(status_code=404, detail=str(error))
fastapi.exceptions.HTTPException

21:53:10,531 src.helpers.handle_error_decorator - ERROR - Error Occurred: get_courses Method Error: 
22:03:07,306 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002C008BD1750>}).
22:03:07,328 src.helpers.setup_logger - DEBUG - get_courses returned cannot access local variable 'fields' where it is not associated with a value.
22:03:43,196 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000024154951710>}).
22:03:43,232 src.helpers.setup_logger - DEBUG - get_courses returned 1.
22:04:08,387 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002D952EBD7D0>}).
22:04:08,411 src.helpers.setup_logger - DEBUG - get_courses returned 1.
22:04:44,218 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000013B2AA3D3D0>}).
22:04:44,239 src.helpers.setup_logger - DEBUG - get_courses returned 'name'.
22:20:05,52 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001E3586BE850>}).
22:20:05,55 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001E3586BEB50>}).
22:20:05,111 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:20:05,115 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:20:20,194 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000023C2CD3DF50>}).
22:20:20,228 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:20:43,533 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001E7C364D6D0>}).
22:20:43,553 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:20:59,312 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001B37318FBD0>}).
22:20:59,339 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:21:20,312 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001E6868F1F10>}).
22:21:20,332 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:21:40,248 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001BFCC8C1D90>}).
22:21:40,284 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:22:02,866 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000020A4EBF2B90>}).
22:22:02,887 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:25:19,52 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002A4E3391910>}).
22:25:19,68 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:26:30,151 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001CCB8FA1DD0>}).
22:26:30,170 src.helpers.setup_logger - DEBUG - get_courses returned Name.
22:27:10,242 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001CAC3F3D6D0>}).
22:27:10,259 src.helpers.setup_logger - DEBUG - get_courses returned Name.
09:20:40,436 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000016FA4989550>}).
09:20:40,441 src.helpers.handle_error_decorator - DEBUG - Error Occurred: get_courses Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 19, in extract_token_data
    payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORTIHM])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 157, in decode
    _validate_claims(
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 481, in _validate_claims
    _validate_exp(claims, leeway=leeway)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 314, in _validate_exp
    raise ExpiredSignatureError("Signature has expired.")
jose.exceptions.ExpiredSignatureError: Signature has expired.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\setup_logger.py", line 22, in wrapper
    val = fun(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 48, in get_courses
    jwt_token_data = extract_token_data(request)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 29, in extract_token_data
    raise HTTPException(status_code=404, detail=str(error))
fastapi.exceptions.HTTPException

09:20:40,441 src.helpers.handle_error_decorator - ERROR - Error Occurred: get_courses Method Error: 
09:46:35,772 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000021319180F90>}).
09:46:35,802 src.helpers.setup_logger - DEBUG - get_courses returned Name.
09:46:47,865 src.helpers.setup_logger - DEBUG - mentor_earning called with ((), {'request': <starlette.requests.Request object at 0x0000021319181310>}).
09:46:47,919 src.helpers.setup_logger - DEBUG - mentor_earning returned [{'name': 'abhishek', 'course_name': 'Master dsa', 'earning': 3000}, {'name': 'abhishek', 'course_name': 'Java script', 'earning': 32000}, {'name': 'abhishek', 'course_name': 'clean code', 'earning': 3000}, {'name': 'abhishek', 'course_name': 'Python', 'earning': 1500}, {'name': 'abhishek', 'course_name': 'dsa master', 'earning': 0}, {'name': 'abhishek', 'course_name': 'dsa ', 'earning': 0}].
09:55:59,665 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000021439FEE210>}).
09:55:59,682 src.helpers.setup_logger - DEBUG - get_courses returned 1.
09:56:15,301 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000194F0EAE490>}).
09:56:15,323 src.helpers.setup_logger - DEBUG - get_courses returned [{}, {}, {}, {}, {}, {}].
09:56:53,738 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002253A914C90>}).
09:56:53,759 src.helpers.setup_logger - DEBUG - get_courses returned [{}, {}, {}, {}, {}, {}].
09:57:08,137 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002860725DF90>}).
09:57:08,152 src.helpers.setup_logger - DEBUG - get_courses returned [{}, {}, {}, {}, {}, {}].
09:58:25,341 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000218EAC5DF50>}).
09:58:25,357 src.helpers.setup_logger - DEBUG - get_courses returned 'name'.
09:58:45,503 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000143DA24E110>}).
09:58:45,518 src.helpers.setup_logger - DEBUG - get_courses returned 'name'.
09:59:16,41 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000220DF5AE210>}).
09:59:16,60 src.helpers.setup_logger - DEBUG - get_courses returned 'name'.
10:00:13,227 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000023B4CAADF50>}).
10:00:13,247 src.helpers.setup_logger - DEBUG - get_courses returned 'name'.
10:01:17,761 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x0000020DFC05DD90>}).
10:01:17,785 src.helpers.setup_logger - DEBUG - get_courses returned 'name'.
10:21:24,564 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002644C70E9D0>}).
10:21:24,598 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 1, 'duration': 'Master dsa', 'price': 'dsa is awesome', 'rating': 12, 'status': 1000, 'approval status': 4.65}, {'name': 3, 'duration': 'Java script', 'price': 'java script is amazing', 'rating': 20, 'status': 2000, 'approval status': 4.79}, {'name': 18, 'duration': 'clean code', 'price': 'dsa is easy and awesome', 'rating': 12, 'status': 1000, 'approval status': 4.75}, {'name': 19, 'duration': 'Python', 'price': "let's talk about variables in python", 'rating': 36, 'status': 1500, 'approval status': 0.0}, {'name': 30, 'duration': 'dsa master', 'price': "let's talk about variables in DSA", 'rating': 12, 'status': 2000, 'approval status': 0.0}, {'name': 31, 'duration': 'dsa ', 'price': "let's talk about variables in DSA", 'rating': 12, 'status': 2000, 'approval status': 0.0}].
10:30:08,279 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001D2DB9AD910>}).
10:30:08,302 src.helpers.setup_logger - DEBUG - get_courses returned unhashable type: 'list'.
10:30:27,765 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002E76400DB90>}).
10:30:27,788 src.helpers.setup_logger - DEBUG - get_courses returned unhashable type: 'list'.
10:30:51,32 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000214DBA5DE10>}).
10:30:51,70 src.helpers.setup_logger - DEBUG - get_courses returned list indices must be integers or slices, not str.
10:49:30,861 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001C9716BF9D0>}).
10:49:30,872 src.helpers.handle_error_decorator - DEBUG - Error Occurred: get_courses Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 19, in extract_token_data
    payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORTIHM])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 157, in decode
    _validate_claims(
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 481, in _validate_claims
    _validate_exp(claims, leeway=leeway)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 314, in _validate_exp
    raise ExpiredSignatureError("Signature has expired.")
jose.exceptions.ExpiredSignatureError: Signature has expired.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\setup_logger.py", line 22, in wrapper
    val = fun(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 48, in get_courses
    jwt_token_data = extract_token_data(request)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 29, in extract_token_data
    raise HTTPException(status_code=404, detail=str(error))
fastapi.exceptions.HTTPException

10:49:30,872 src.helpers.handle_error_decorator - ERROR - Error Occurred: get_courses Method Error: 
10:49:48,721 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000184A153E150>}).
10:49:48,727 src.helpers.handle_error_decorator - DEBUG - Error Occurred: get_courses Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 19, in extract_token_data
    payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORTIHM])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 157, in decode
    _validate_claims(
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 481, in _validate_claims
    _validate_exp(claims, leeway=leeway)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 314, in _validate_exp
    raise ExpiredSignatureError("Signature has expired.")
jose.exceptions.ExpiredSignatureError: Signature has expired.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 75, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\setup_logger.py", line 22, in wrapper
    val = fun(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py", line 48, in get_courses
    jwt_token_data = extract_token_data(request)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 29, in extract_token_data
    raise HTTPException(status_code=404, detail=str(error))
fastapi.exceptions.HTTPException

10:49:48,727 src.helpers.handle_error_decorator - ERROR - Error Occurred: get_courses Method Error: 
10:49:59,822 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000184A15578D0>}).
10:49:59,842 src.helpers.setup_logger - DEBUG - get_courses returned unhashable type: 'list'.
10:50:09,201 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000199159ADC10>}).
10:50:09,221 src.helpers.setup_logger - DEBUG - get_courses returned unhashable type: 'list'.
10:50:32,275 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001C3A2D5DF10>}).
10:50:32,333 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 1, 'duration': 'Master dsa', 'price': 'dsa is awesome', 'rating': 12, 'status': 1000, 'approval status': 4.65}, {'name': 3, 'duration': 'Java script', 'price': 'java script is amazing', 'rating': 20, 'status': 2000, 'approval status': 4.79}, {'name': 18, 'duration': 'clean code', 'price': 'dsa is easy and awesome', 'rating': 12, 'status': 1000, 'approval status': 4.75}, {'name': 19, 'duration': 'Python', 'price': "let's talk about variables in python", 'rating': 36, 'status': 1500, 'approval status': 0.0}, {'name': 30, 'duration': 'dsa master', 'price': "let's talk about variables in DSA", 'rating': 12, 'status': 2000, 'approval status': 0.0}, {'name': 31, 'duration': 'dsa ', 'price': "let's talk about variables in DSA", 'rating': 12, 'status': 2000, 'approval status': 0.0}].
10:50:46,466 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001C3A3121890>}).
10:50:46,513 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 1, 'duration': 'Master dsa', 'price': 'dsa is awesome', 'rating': 12, 'status': 1000, 'approval status': 4.65}, {'name': 3, 'duration': 'Java script', 'price': 'java script is amazing', 'rating': 20, 'status': 2000, 'approval status': 4.79}, {'name': 18, 'duration': 'clean code', 'price': 'dsa is easy and awesome', 'rating': 12, 'status': 1000, 'approval status': 4.75}, {'name': 19, 'duration': 'Python', 'price': "let's talk about variables in python", 'rating': 36, 'status': 1500, 'approval status': 0.0}, {'name': 30, 'duration': 'dsa master', 'price': "let's talk about variables in DSA", 'rating': 12, 'status': 2000, 'approval status': 0.0}, {'name': 31, 'duration': 'dsa ', 'price': "let's talk about variables in DSA", 'rating': 12, 'status': 2000, 'approval status': 0.0}].
10:59:47,511 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002469D07E850>}).
10:59:47,548 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Master dsa', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.65, 'approval_status': 'approved', 'status': 'deactive'}, {'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79, 'approval_status': 'approved', 'status': 'active'}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75, 'approval_status': 'approved', 'status': 'active'}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0, 'approval_status': 'approved', 'status': 'active'}, {'name': 'dsa master', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'approval_status': 'pending', 'status': 'active'}, {'name': 'dsa ', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'approval_status': 'pending', 'status': 'active'}].
11:00:01,139 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002469D07E8D0>}).
11:00:01,172 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Master dsa', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.65, 'approval_status': 'approved', 'status': 'deactive'}, {'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79, 'approval_status': 'approved', 'status': 'active'}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75, 'approval_status': 'approved', 'status': 'active'}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0, 'approval_status': 'approved', 'status': 'active'}, {'name': 'dsa master', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'approval_status': 'pending', 'status': 'active'}, {'name': 'dsa ', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'approval_status': 'pending', 'status': 'active'}].
11:00:14,165 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002469D0C3A50>}).
11:00:14,175 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
11:02:21,350 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002469D0C0A50>}).
11:02:21,359 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
11:12:57,608 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000268E22ADE10>}).
11:12:57,617 src.helpers.setup_logger - DEBUG - get_courses returned name 'list_course_role_1' is not defined.
11:13:00,913 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000268E22C7D10>}).
11:13:00,921 src.helpers.setup_logger - DEBUG - get_courses returned name 'list_course_role_1' is not defined.
11:14:55,120 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x00000268E22EE110>}).
11:14:55,153 src.helpers.setup_logger - DEBUG - get_courses returned name 'list_course_role_1' is not defined.
11:15:12,540 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002B27D37E2D0>}).
11:15:12,575 src.helpers.setup_logger - DEBUG - get_courses returned name 'list_course_role_1' is not defined.
11:15:56,545 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001FA24E1DD10>}).
11:15:56,565 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Master dsa', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.65, 'approval_status': 'approved', 'status': 'deactive'}, {'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79, 'approval_status': 'approved', 'status': 'active'}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75, 'approval_status': 'approved', 'status': 'active'}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0, 'approval_status': 'approved', 'status': 'active'}, {'name': 'dsa master', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'approval_status': 'pending', 'status': 'active'}, {'name': 'dsa ', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'approval_status': 'pending', 'status': 'active'}].
11:16:08,827 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001FA24E3BE90>}).
11:16:08,837 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
11:25:33,983 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002688BBD1810>}).
11:25:34,1 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
11:25:44,124 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000002688BC5E7D0>}).
11:25:44,135 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
11:28:25,276 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001F45051D950>}).
11:28:25,286 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
11:28:46,201 src.helpers.setup_logger - DEBUG - mentor_earning called with ((), {'request': <starlette.requests.Request object at 0x000001F45051EE10>}).
11:28:46,305 src.helpers.setup_logger - DEBUG - mentor_earning returned [{'name': 'abhishek', 'course_name': 'Master dsa', 'earning': 3000}, {'name': 'abhishek', 'course_name': 'Java script', 'earning': 32000}, {'name': 'abhishek', 'course_name': 'clean code', 'earning': 3000}, {'name': 'abhishek', 'course_name': 'dsa master', 'earning': 0}, {'name': 'abhishek', 'course_name': 'dsa ', 'earning': 0}].
11:28:58,680 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001F45051DE90>}).
11:28:58,689 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
11:29:11,352 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001F45051F3D0>}).
11:29:11,361 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
11:29:33,496 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001F450522BD0>}).
11:29:33,523 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Master dsa', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.65, 'approval_status': 'approved', 'status': 'deactive'}, {'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79, 'approval_status': 'approved', 'status': 'active'}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75, 'approval_status': 'approved', 'status': 'active'}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0, 'approval_status': 'approved', 'status': 'active'}, {'name': 'dsa master', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'approval_status': 'pending', 'status': 'active'}, {'name': 'dsa ', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0, 'approval_status': 'pending', 'status': 'active'}].
11:32:06,962 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001344684EC10>}).
11:32:06,980 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
13:40:57,880 src.helpers.setup_logger - DEBUG - add_course called with ((), {'body': {'name': 'dsa', 'content': "let's talk about variables in DSA", 'price': 2000, 'duration': 12}, 'request': <starlette.requests.Request object at 0x0000029DAA735710>}).
13:40:57,928 src.helpers.setup_logger - DEBUG - add_course returned {'message': 'Course approval request sent to admin.'}.
13:43:27,686 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x000002231932B7D0>}).
13:43:27,701 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:43:27,701 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:44:06,564 src.helpers.setup_logger - DEBUG - add_course called with ((), {'body': {'name': 'namaste react ', 'content': 'React is a beast.', 'price': 2000, 'duration': 12}, 'request': <starlette.requests.Request object at 0x0000022319351D10>}).
13:44:06,594 src.helpers.setup_logger - DEBUG - add_course returned {'message': 'Course approval request sent to admin.'}.
13:44:20,339 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x0000022319351950>}).
13:44:20,350 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:44:20,350 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:45:03,975 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x000001D87A8FD390>}).
13:45:03,991 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:45:03,991 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:45:47,403 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x0000029F9A81D8D0>}).
13:45:47,421 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:45:47,421 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:45:48,182 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x0000029F9A83B490>}).
13:45:48,189 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:45:48,190 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:49:32,4 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x0000024D1010DBD0>}).
13:49:32,20 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:49:32,20 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:49:56,799 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x0000024D10510590>}).
13:49:56,808 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:49:56,808 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:50:29,984 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x0000022FFC0BDC90>}).
13:50:29,994 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:50:29,994 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:50:49,807 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x000001E097AFD950>}).
13:50:49,817 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:50:49,818 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:52:29,96 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x000001603CDBDDD0>}).
13:52:29,107 root - ERROR - 2055: Lost connection to MySQL server at 'localhost:3306', system error: Connection not available.
13:52:29,107 src.helpers.setup_logger - DEBUG - pending_courses returned {'message': 'No pending course'}.
13:53:58,525 src.helpers.setup_logger - DEBUG - pending_courses called with ((), {'request': <starlette.requests.Request object at 0x00000217D464DBD0>}).
13:53:58,534 src.helpers.setup_logger - DEBUG - pending_courses returned [{'name': 'dsa master', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0}, {'name': 'dsa ', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0}, {'name': 'dsa', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0}, {'name': 'namaste react ', 'duration (in hrs.)': 12, 'price (in Rs.)': 2000, 'rating': 0.0}].
14:14:27,390 src.helpers.setup_logger - DEBUG - add_course called with ((), {'body': {'name': 'python', 'content': 'React is a beast.', 'price': 2000, 'duration': 12}, 'request': <starlette.requests.Request object at 0x000001C38A589690>}).
14:14:27,405 src.helpers.setup_logger - DEBUG - add_course returned {'message': 'Another course with the same name already exists. Please try again.'}.
15:04:27,787 src.helpers.setup_logger - DEBUG - purchase_course called with ((), {'course_name': 'clean code', 'request': <starlette.requests.Request object at 0x000001BE3FD512D0>}).
15:04:27,805 src.helpers.setup_logger - DEBUG - purchase_course returned {'message': "You've already purchased this course."}.
15:04:31,517 src.helpers.setup_logger - DEBUG - get_courses called with ((), {'request': <starlette.requests.Request object at 0x000001BE3FD51650>}).
15:04:31,525 src.helpers.setup_logger - DEBUG - get_courses returned [{'name': 'Java script', 'duration (in hrs.)': 20, 'price (in Rs.)': 2000, 'rating': 4.79}, {'name': 'clean code', 'duration (in hrs.)': 12, 'price (in Rs.)': 1000, 'rating': 4.75}, {'name': 'Python', 'duration (in hrs.)': 36, 'price (in Rs.)': 1500, 'rating': 0.0}].
15:05:17,503 src.helpers.setup_logger - DEBUG - mentor_earning called with ((), {'request': <starlette.requests.Request object at 0x000001BE3FD55790>}).
15:05:17,571 src.helpers.setup_logger - DEBUG - mentor_earning returned [{'name': 'abhishek', 'course_name': 'Master dsa', 'earning': 3000}, {'name': 'abhishek', 'course_name': 'Java script', 'earning': 32000}, {'name': 'abhishek', 'course_name': 'clean code', 'earning': 3000}, {'name': 'abhishek', 'course_name': 'dsa master', 'earning': 0}, {'name': 'abhishek', 'course_name': 'dsa ', 'earning': 0}, {'name': 'abhishek', 'course_name': 'dsa', 'earning': 0}, {'name': 'abhishek', 'course_name': 'namaste react ', 'earning': 0}].
15:13:28,246 src.helpers.handle_error_decorator - DEBUG - Error Occurred: login_user Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 38, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\login_signup.py", line 64, in login_user
    is_authenticated = login.login_user(
                       ^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\auth.py", line 31, in login_user
    user_data = self.validate_credentials(self.username, self.password)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\auth.py", line 83, in validate_credentials
    response = db.get_from_db(QUERIES.get("GET_FROM_AUTHENTICATION"), (username,))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\models\database.py", line 69, in get_from_db
    cursor = db.cursor()
             ^^^^^^^^^^^
TypeError: 'CMySQLCursor' object is not callable

15:13:28,246 src.helpers.handle_error_decorator - ERROR - Error Occurred: login_user Method Error: 'CMySQLCursor' object is not callable
16:02:45,668 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 38, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 29, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 37, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:02:45,668 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
14:03:27,644 src.helpers.setup_logger - INFO - function called admin
14:07:01,664 src.helpers.setup_logger - INFO - C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py - get_courses at line 51: function called admin
14:08:11,436 src.helpers.setup_logger - DEBUG - C:\coding\WG\Online_Learning_System - Fast\src\routes\courses.py - get_courses at line 51: function called admin
16:59:07,994 src.helpers.handle_error_decorator - DEBUG - Error Occurred: purchase_courses Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 19, in extract_token_data
    payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORTIHM])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 157, in decode
    _validate_claims(
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 481, in _validate_claims
    _validate_exp(claims, leeway=leeway)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\jose\jwt.py", line 314, in _validate_exp
    raise ExpiredSignatureError("Signature has expired.")
jose.exceptions.ExpiredSignatureError: Signature has expired.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 17, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\students.py", line 13, in purchase_courses
    jwt_token_data = extract_token_data(request)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\jwt_helpers.py", line 29, in extract_token_data
    raise HTTPException(status_code=404, detail=str(error))
fastapi.exceptions.HTTPException

16:59:07,994 src.helpers.handle_error_decorator - ERROR - Error Occurred: purchase_courses Method Error: 
21:51:36,512 src.helpers.setup_logger - DEBUG - function called admin
22:41:42,821 src.helpers.setup_logger - DEBUG - function called admin
15:59:40,921 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 29, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 38, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

15:59:40,921 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:01:19,911 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 29, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 38, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:01:19,911 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:01:36,620 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 29, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 38, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:01:36,620 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:01:53,720 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 30, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 38, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:01:53,720 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:02:15,349 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 31, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 38, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:02:15,349 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:02:38,710 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 32, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 38, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:02:38,710 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:03:00,867 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 32, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 39, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:03:00,867 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:03:13,910 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 32, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 40, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:03:13,910 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:03:28,406 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 32, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 41, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:03:28,406 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:04:05,827 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 32, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 41, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:04:05,828 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:04:06,740 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 32, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 22, in add_mentor
    is_valid_username = db.get_from_db(
                        ^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\models\database.py", line 57, in get_from_db
    self.cursor.execute(query, val)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\mysql\connector\cursor_cext.py", line 303, in execute
    self._cnx.handle_unread_result()
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\mysql\connector\connection_cext.py", line 955, in handle_unread_result
    raise InternalError("Unread result found")
mysql.connector.errors.InternalError: Unread result found

16:04:06,740 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: Unread result found
16:04:17,883 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 32, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 22, in add_mentor
    is_valid_username = db.get_from_db(
                        ^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\models\database.py", line 57, in get_from_db
    self.cursor.execute(query, val)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\mysql\connector\cursor_cext.py", line 303, in execute
    self._cnx.handle_unread_result()
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\mysql\connector\connection_cext.py", line 955, in handle_unread_result
    raise InternalError("Unread result found")
mysql.connector.errors.InternalError: Unread result found

16:04:17,883 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: Unread result found
16:04:51,48 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 32, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 41, in add_mentor
    (Roles.MENTOR.value, result[0][3]),
                         ~~~~~~^^^
TypeError: 'int' object is not subscriptable

16:04:51,48 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: 'int' object is not subscriptable
16:06:43,178 src.helpers.handle_error_decorator - DEBUG - Error Occurred: add_new_mentor Method Error: Traceback (most recent call last):
  File "C:\coding\WG\Online_Learning_System - Fast\src\helpers\handle_error_decorator.py", line 15, in wrapper
    return_value = function(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\routes\mentors.py", line 32, in add_new_mentor
    message = mentor.add_mentor(username)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\coding\WG\Online_Learning_System - Fast\src\controllers\mentor.py", line 39, in add_mentor
    db.update_db(
  File "C:\coding\WG\Online_Learning_System - Fast\src\models\database.py", line 45, in update_db
    self.cursor.execute(query, val)
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\mysql\connector\cursor_cext.py", line 303, in execute
    self._cnx.handle_unread_result()
  File "C:\Users\anaugaraiya\.virtualenvs\Online_Learning_System_-_Fast-H9MtE_4s\Lib\site-packages\mysql\connector\connection_cext.py", line 955, in handle_unread_result
    raise InternalError("Unread result found")
mysql.connector.errors.InternalError: Unread result found

16:06:43,179 src.helpers.handle_error_decorator - ERROR - Error Occurred: add_new_mentor Method Error: Unread result found
12:16:45,841 src.helpers.handle_error_decorator - ERROR - Invalid Credentials
12:17:43,814 src.helpers.handle_error_decorator - ERROR - Invalid Credentials
12:17:54,611 src.helpers.handle_error_decorator - ERROR - Invalid Credentials
12:17:57,569 src.helpers.handle_error_decorator - ERROR - Invalid Credentials
12:17:59,202 src.helpers.handle_error_decorator - ERROR - Invalid Credentials
12:46:29,74 src.helpers.handle_error_decorator - ERROR - 
